### ISSUE TITLE
Support merges that have null values for some numeric data (int64, float64, etc...)

### ISSUE THREAD

**Issue by JSpenced** [lance-format/lance#1214]
I get an error when trying to merge a column that only has part of the date range included (eg... I don't do an entire backfill). If I had 4 years of data, I only wanted to backfill the column two years then I would get the following:
```OSError: Invalid user input: Found rows on LHS that do not match any rows on RHS. Lance would need to write nulls on the RHS, but Lance does not yet support nulls for type Int32.```

Is there any plan to support Nulls for `Int` types or is there an int type that does support nulls now? I tried different `int` and `float` types including pandas nullable types `Int32`, but nothing seemed to work.

TLDR, Is there a way to do a numeric column merge and have null fields?

**Comment by wjones127** [comment]
Right now the only columns that support nulls in Lance are string, binary, and list columns.

Support for null values in all other types is planned, but likely not for at least a couple of months. It will require a large change to the encodings.

A simpler solution we could do is allow filling a different value if not found (such as `0` or `-1` for int and `NaN` for float). That would be quicker to implement.

**Comment by JSpenced** [comment]
Okay, thanks! Yeah that would help for now so it doesn't error out as that would be what we probably do anyways is specify a fill-value. Honestly, we don't need this now, but I was thinking about 2 years down the road and we don't need to backfill 6 years of data for some use-cases.

To that end, if you update lance and we incorporate a merge with null values after a dataset has already been created could it convert that column to nullable so the merge would work? Or we would have to recreate it? Not really a blocker just curious. 

Thanks for the speedy response.

**Comment by wjones127** [comment]
> To that end, if you update lance and we incorporate a merge with null values after a dataset has already been created could it convert that column to nullable so the merge would work?

Well the new column that would be getting nulls would be new, so there's nothing to recreate, right?

**Comment by JSpenced** [comment]
Yeah, that wasn't phrased well. I mean at a future date once null columns are available they can be merged in even if not supported when the dataset was created.

**Comment by westonpace** [comment]
As an update for future readers.  Nulls should currently be supported on all data types except structs.  Some old datasets may still be using legacy storage.  You can check by printing the data storage version:

```
>>> print(ds.data_storage_version)
0.1 # Legacy storage, will require migration if nulls are desired
```

If this is the case and you want to migrate your data to the new storage version you can do so by making a copy of your dataset:

```
>>> ds = lance.write_dataset(ds, "/tmp/new_ds.lance")
>>> print(ds.data_storage_version)
2.0
```
